bam_to_mate_hist.py

```diff
@@ -241,8 +310,17 @@ def estimate_required_num_reads(diff_contig, refs, num_pairs, target=600.0):
     return int(total_num)
 
 
-def extract_stats(stat_list, bamfile, outfile_name, count_diff_refname_stub=False):
-    '''Make a dict of results and data suitable to be passed to the report template for splatting.
+def est_proportions_pretty(stat_dict, stats=None):
+    '''Compute proportions from the dictionary for specified statistics, make them printable. '''
+    out_dict = stat_dict.copy()
+    num_pairs = stat_dict["NUM_PAIRS"]
+    for stat in stats:
+        out_dict[stat] = "{0:.3f}".format(float(stat_dict[stat]) / num_pairs)
```

Comment:
Do you care if you have small rounding issues?  I believe this will truncate/floor your float value rather than rounding it (i.e. 16.1236 displays as 16.123 instead of 16.124).  It isn't a major accuracy issue for these reports most likely, but it is probably worth using a precision_round call if you care about getting the exact right number.  This is from pgutils.utilities if you want to just add the function to your script:
`python
def precision_round(val, decimal_places):
    '''Round value to the given number of decimal places using the Decimal library

    Args:
        val (float): Value to round
        decimal_places (int): Number of decimal places to round to

    Returns:
        (float): Number rounded to the desired precision
    '''
    places = -1 * decimal_places
    return decimal.Decimal(val).quantize(decimal.Decimal(10) ** places)`

Reply:
i think i don't care? these are just the values displayed on the report.